substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -1979,6 +2085,12 @@ where
 
 		self.view_store.finish_background_revalidations().await;
 
+		// Special handling for Reverted - bypass normal flow
+		if let ChainEvent::Reverted { .. } = &event {
+			self.handle_reverted(&event).await;
+			return;
+		}
+
```

Comment:
We could have `EnactmentAction::HandleReverted`, which could use return the block number and the tree route. It could internally update best_block (and finalized if we agree to do so) - so force_updated could be avoidded.

I think this way we could fit the reverted event nicely/elegantly into the existing flow, w/o extra returns on the path.